warehouse/query-core/src/main/java/datawave/query/jexl/visitors/ExecutableDeterminationVisitor.java

```diff
@@ -149,6 +152,51 @@ public ExecutableDeterminationVisitor(ShardQueryConfiguration conf, MetadataHelp
         }
     }
     
+    /**
+     * Negate the current data object
+     * 
+     * @param data
+     *            the data passed along to hte visitor, may be null or a string
+     * @return NEGATION_PREFIX if the string was null or the old string appended with NEGATION_PREFIX
+     */
+    public static Object negateData(Object data) {
```

Comment:
Why don't we simple send a boolean instead of a string, and negate the boolean when we hit a negation?  Using a string that we have to parse seems kinda odd.

Reply:
this is a legacy artifact from the string representation that I didn't want to disrupt with this ticket.

Reply:
This seemed like the most painless way to integrate with the current output logging without also revamping how that was done.

Reply:
sigh.... ok